### DISCUSSION TITLE
Installation of V1.12 with Minikube

### DISCUSSION THREAD

**Discussion by choonho** [cloudforet-io/community#134] [Q&A]
# Installation Guide

https://cloudforet.io/docs/setup_operation/quick_install/

If you have any issues about installation with Minikube, Please ask here!

**Comment by arup1221** [comment]
as mentioned ARM64 architecture (We do not support minikube with ARM64 CPU, such as MacBook M2)

Does in macbook m1 will it support?

**Reply by choonho** [reply]
This documentation is written at version 1.11.

Cloudforet supports ARM Docker image from version 1.12.

Please try Macbook. No one tested yet.

**Reply by arup1221** [reply]
ok, I will try.

**Reply by arup1221** [reply]
@choonho  in macbook M1 it installed successfully

**Comment by achiverram28** [comment]
So Macbook M2 users cannot use?

**Reply by achiverram28** [reply]
Btw , the current helm release version is v3.13.2 , but in the document , it is 3.2.0+

**Reply by achiverram28** [reply]
Yup , we do have an error in macbook m2, am I the only one facing it , or anyone else too?

**Reply by arup1221** [reply]
do you have docker desktop installed?

**Reply by choonho** [reply]
I also have M2 Macbook. I succeeded Cloudforet installation.
If your Helm version is 3+, it will work.

**Reply by choonho** [reply]
Helm 3.2.0+ is typo. Please update documentation. Anyone can make Pull Request.

The verified Helm version is over 3.11.0+

**Reply by achiverram28** [reply]
Yup I am making the PR

**Reply by achiverram28** [reply]
Yes I have docker desktop.
I will check once more

**Comment by achiverram28** [comment]
https://github.com/cloudforet-io/docs/pull/168

Check this PR , for the typo correct

**Comment by achiverram28** [comment]
Was able to install correctly

**Comment by Asfawh** [comment]
I am kinda confused with the first Prerequisites instruction that we need an AWS EC2 VM (Intel/AMD/ARM CPU), is there an aws account somewhere or are we using our aws account and money for the project? if not I should have a dedicated on machine for k8s cluster or minikube along with other tools?

**Reply by Asfawh** [reply]
I wish if there is a separate slack channel for discussion, and I wasn't aware if there was a zoom introduction

**Reply by Asfawh** [reply]
tried install CloudForet on my personal macbk following instruction https://cloudforet.io/docs/setup_operation/installation/on_premise/, first of all initialize-spaceone-v2iro-sz6rq:  status  0/1 Completed. And checked its log see 'errmsg': 'Authentication failed' not sure what what step I missed. 

https://gist.github.com/Asfawh/aaeb88385e0fadc35e0a3c75d5973a7e

**Reply by choonho** [reply]
I found one possible scenario. Did your create "initiailizer.yaml" file using following instruction?

![image](https://github.com/cloudforet-io/cloudforet/assets/3993088/5c07a7ba-b662-45e9-9ae6-152e9d7b860f)

This has some typo with wrong indentation. Please compare with 
https://github.com/cloudforet-io/charts/blob/master/examples/initializer.yaml

I am highly recommend using https://github.com/cloudforet-io/charts/blob/master/examples/initializer.yaml

I also create new issue which you can create PR. Anyone is welcome.

https://github.com/cloudforet-io/docs/issues/173

**Comment by achiverram28** [comment]
> I am kinda confused with the first Prerequisites instruction that we need an AWS EC2 VM (Intel/AMD/ARM CPU), is there an aws account somewhere or are we using our aws account and money for the project? if not I should have a dedicated on machine for k8s cluster or minikube along with other tools?

As of now , in the week 1 we are focussing on minikube setup and not on any cloud provider
Kindly checkout -> https://github.com/orgs/cloudforet-io/discussions/133 for the process flow

**Comment by ATREAY** [comment]
Hey everyone, does anyone know how to set up cloudforet on Windows? 
I'm running into a bunch of errors, and you can check them out [here](https://github.com/orgs/cloudforet-io/discussions/133#discussioncomment-7692232).

**Comment by justankiit** [comment]
Hey @choonho, I have tried to setup the cloudforet on ec2 instance and almost completed but I got confused & error at 6th step where we have to create a values.yaml and edit it too!!
When I ran on ec2 it has space of 3847 Mb, this projects need is 4096 Mb, how can I achieve it!!
Can you create a Installation Video of this Project on YT So, Many folks can understand and setup it easily!!


![image](https://github.com/cloudforet-io/cloudforet/assets/111564362/e09f929b-3d57-4566-9608-b7f5ccfffff7)

**Reply by ATREAY** [reply]
It would be great if @choonho or anyone who has successfully set up cloudforet locally could create a video tutorial. Many mentees are having errors, and a video would be really helpful.

**Reply by ATREAY** [reply]
Hey @kumarankit999, did you manage to set up locally?

**Reply by justankiit** [reply]
Hey @ATREAY, I'm working and setup on ec2 instance of AWS!

**Comment by 0xPunkItachi** [comment]
i followed the cmds step by step mentioned in the docs and in the step 2 i'm getting this error, i was trying to fix it by my own but unable to so. pls help if possible
![Screenshot 2023-12-02 100928](https://github.com/cloudforet-io/cloudforet/assets/140940344/bc42bb98-1718-4145-b34e-e9a5122c3166)

**Reply by sundarr786** [reply]
I am not sure why this error comes, make Sure you installed Prerequisites 
and run minikube start --driver=docker --memory=4096mb before step 1 as mentioned in Documentation

and try with command prefix with sudo

**Reply by 0xPunkItachi** [reply]
hii thanks mate, actually the above issue is resolved now but 🥲🥲🥲.....................................................................................
There is a new error. I am working on this installation from 27 nov and still its not done. I dont know what to do now

![Screenshot (158)](https://github.com/cloudforet-io/cloudforet/assets/140940344/3dcd88d2-bb46-414b-9e28-1901bb429864)

pls someone help if possible @choonho

**Reply by Gaurav122000** [reply]
@YashCommits what's the error and on which step you are on

**Reply by 0xPunkItachi** [reply]
Every pod must show running status but in my case it's not. It is showing crash loop back off... So if i proceed without rectifying it will throw error in future somewhere.

BTW can you tell me that the due date of task is tomorrow after 11:59pm right ??? @Gaurav122000 

Also @Gaurav122000 can you connect with me in Twitter? My username is @YashCommits in Twitter... I am asking for this so we can learn together and resolve our problems

**Reply by Gaurav122000** [reply]
@YashCommits pod takes some time to run try " kubectl get pods -n spaceone "once again 2 to 3 times if still not running then their a problem and you didn't tell on which step you are at.
I don't know the due date.
we can connect to LinkedIn - https://www.linkedin.com/in/gaurav-sharma-083786274/

**Reply by 0xPunkItachi** [reply]
Alright bro its step no. 6. And I have sent the connection on LinkedIn

**Reply by Gaurav122000** [reply]
@YashCommits delete your minikube from the docker and restart the docker then follow the step 1 to 6 again.

**Reply by Gaurav122000** [reply]
@YashCommits I accepted the invitation

**Comment by Gaurav122000** [comment]
Hey did any one found a solution for window 404 error

**Comment by justankiit** [comment]
Hey @choonho, where is the Maintainer/Mentor of this Mentorship Program!!?????????
Why he is not helping folks here?

**Comment by 0xPunkItachi** [comment]
How much time it took in your case to show running or completed status.... i am waiting for this for last 1 and a half hours

![Screenshot 2023-12-03 152129](https://github.com/cloudforet-io/cloudforet/assets/140940344/64a782b2-cc4a-4a0b-aea8-e830d642dbda)

**Reply by 0xPunkItachi** [reply]
Hello @choonho 👋
Sir I tried installation multiple times  in different ways

1. ubantu(Virtual box)
2. WSL2
3. GitHub codespaces

I'm not lying sir but I keep getting error in step no. 6 (some pods status were showing crashloopbackoff while some were showing creating container & i waited for it like 2 hours... still their status didn't changed from creating container to running)

I'm a newbie in this space and I'm learning docker and kubernetes. (Feeling Lil bit nervous to ask for help bcz it maybe a silly doubt to ask)

If anyone has installed it locally, pls help me in fixing it & pls reply asap.

**Reply by choonho** [reply]
@YashCommits @Gaurav122000 @Asfawh   I created 1:1 google calendar. Please reserve time-slot what you can. I will help you.

Aynone who has problem to install, please reserve google calendar.

https://calendar.google.com/calendar/u/0?cid=MTdlYzU2MjFjNDlhN2Y3ZGFlMzc1NWJhM2EwNjhkZmNhNDA2NTY0OGMxYTM4ZGZmMWE3MWViZjlmMGIwZDU1NUBncm91cC5jYWxlbmRhci5nb29nbGUuY29t

**Reply by justankiit** [reply]
Have you created a time slot, or we have to create it @choonho!

**Reply by choonho** [reply]
@kumarankit999 In the calendar, time-slots are already made. Update title what you want.

**Reply by choonho** [reply]
I think you can't update. Please ***add comment what you want*** at https://github.com/orgs/cloudforet-io/discussions/136

**Reply by choonho** [reply]
@YashCommits You executed initialization, before finishing the installation. 

I updated docs.

![image](https://github.com/cloudforet-io/cloudforet/assets/3993088/d2fc4e94-ab3c-4436-8b83-0246a07bbc36)

**Comment by aquilinux** [comment]
Hello, i have this error trying to get TOKEN:

```
START SPACEONE INITIALIZE
TASK [Create Domain] ***************************************************************************************************
Start identity.Domain.create

	error_code = ERROR_GRPC_CONNECTION
	status_code = UNAVAILABLE
	message = Server is unavailable. (channel = identity:50051, message = Channel is not ready.)

(ERROR) =>

	error_code = ERROR_GRPC_CONNECTION
	status_code = UNAVAILABLE
	message = Server is unavailable. (channel = identity:50051, message = Channel is not ready.)
>> Spec: {"resource_type": "identity.Domain", "data": {"name": "root", "config": {"LANGUAGE": "ko", "TIMEZONE": "Asia/Seoul"}}, "matches": ["name"], "mode": "NO_UPDATE", "verb": {"read": "list", "create": "create", "update": "update"}, "no_verification": true}
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/spaceone/core/pygrpc/client.py", line 447, in client
    grpc.channel_ready_future(channel).result(timeout=3)
  File "/usr/local/lib/python3.8/site-packages/grpc/_utilities.py", line 162, in result
    self._block(timeout)
  File "/usr/local/lib/python3.8/site-packages/grpc/_utilities.py", line 106, in _block
    raise grpc.FutureTimeoutError()
grpc.FutureTimeoutError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/spacectl-0.0.0-py3.8.egg/spacectl/lib/apply/task.py", line 19, in wrapper
    func(self)
  File "/usr/local/lib/python3.8/site-packages/spacectl-0.0.0-py3.8.egg/spacectl/modules/resource/__init__.py", line 76, in execute
    self._create(service, resource)
  File "/usr/local/lib/python3.8/site-packages/spacectl-0.0.0-py3.8.egg/spacectl/modules/resource/__init__.py", line 145, in _create
    create_result = _execute_api(service, resource, verb, self.spec['data'], api_version="v1", output="yaml", parser={}, silent=self.silent)
  File "/usr/local/lib/python3.8/site-packages/spacectl-0.0.0-py3.8.egg/spacectl/modules/resource/__init__.py", line 282, in _execute_api
    client = _get_client(service, api_version)
  File "/usr/local/lib/python3.8/site-packages/spacectl-0.0.0-py3.8.egg/spacectl/command/execute.py", line 271, in _get_client
    raise e
  File "/usr/local/lib/python3.8/site-packages/spacectl-0.0.0-py3.8.egg/spacectl/command/execute.py", line 265, in _get_client
    client = pygrpc.client(endpoint=f'{e.get("hostname")}:{e.get("port")}',
  File "/usr/local/lib/python3.8/site-packages/spaceone/core/pygrpc/client.py", line 449, in client
    raise ERROR_GRPC_CONNECTION(channel=endpoint, message='Channel is not ready.')
spaceone.core.error.ERROR_GRPC_CONNECTION: 
	error_code = ERROR_GRPC_CONNECTION
	status_code = UNAVAILABLE
	message = Server is unavailable. (channel = identity:50051, message = Channel is not ready.)

FINISH SPACEONE INITIALIZE
```
Thanks for helping

**Reply by mennatallah-abdallah-e** [reply]
How did you fix this error? I am facing the same.

**Comment by choonho** [comment]
@aquilinux ***Channel is not ready*** means docker container is not running. Initialization must be finished after all POD is ***running***.

**Reply by aquilinux** [reply]
thanks. fixed

**Comment by aquilinux** [comment]
Hello, today i have 404 error instead of login page but when i last checked past week all was working as expected (i worked on the site with no issues).
Kind cluster running on linux, all pods running, port forwarding in place.
What is 404 error originated from?

**Reply by choonho** [reply]
***  5. Initialize the Configuration *** is a database operation. If you failed initialization, then execute again, it failed to initialize.
I highly recommend re-install after clean-up all. Usually 404 means frontend service is working correctly, but backend services is not working.

**Comment by Tyagiquamar** [comment]
Anyone here, I tried to install again it on manjaro but 
minikube start --vm-driver=docker --memory=4096                                               ✔ 

Starting local Kubernetes cluster...
F0117 11:54:23.904072   61207 cluster.go:391] Unsupported driver: docker

**Comment by lisinan88** [comment]
Dear  All 

why i get  404

![image](https://github.com/cloudforet-io/cloudforet/assets/129405455/a7c78f0a-e702-4f0f-a0c9-c1abd66d3f3b)

**Reply by mennatallah-abdallah-e** [reply]
the values.yaml file needs to have the correct console api and api2 urls. @lisinan88

**Reply by lisinan88** [reply]
i follow this document ,do i need more step to install in aws ec2?

**Reply by lisinan88** [reply]
![image](https://github.com/cloudforet-io/cloudforet/assets/129405455/0bb1f78c-5873-45e0-9e21-d204827f2c8f)
i just change the endpoint

**Reply by mennatallah-abdallah-e** [reply]
I faced the 404 issue while installing on AWS and I updated the endpoints in this step, they were incorrect for me .

https://cloudforet.io/docs/setup_operation/installation/aws/#6-set-the-helm-values-and-upgrade-the-chart

**Reply by mennatallah-abdallah-e** [reply]
@lisinan88  have you tried these two endpoints from browser?

**Reply by lisinan88** [reply]
yes  it also not work for me

**Reply by mennatallah-abdallah-e** [reply]
that's the issue, would you please share screenshot for "kubectl get svc -n spaceone"

**Reply by lisinan88** [reply]
![image](https://github.com/cloudforet-io/cloudforet/assets/129405455/8e6f76eb-4401-4064-8dce-75169576768a)
here it is

**Reply by lisinan88** [reply]
i  did not do 

kubectl delete po -n spaceone -l app.kubernetes.io/instance=cloudforet

**Reply by mennatallah-abdallah-e** [reply]
the console-api is exposed using nodeport which is accessible on http://localhost:31952
Can u test this url  http://localhost:31952

**Comment by mennatallah-abdallah-e** [comment]
I am getting "Failed to load additional options." whenever I try to create a collector. Anyone faced this ?

**Reply by choonho** [reply]
Usually, console-api-v2 is not reachable. Check the URL.

* CONSOLE_API_V2 should be reachable from your browser.

```
console:
  production_json:
    CONSOLE_API:
      ENDPOINT: http://localhost:8081  # http://ec2_public_ip:8081 for EC2 users
    CONSOLE_API_V2:
      ENDPOINT: http://localhost:8082  # http://ec2_public_ip:8082 for EC2 users

global:
  shared_conf:
    TOKEN: 'TOKEN_VALUE_FROM_ABOVE'   # Change the system token```
```

**Reply by lisinan88** [reply]
yes  i get this issue resolved

but ,just a little collector to use

**Reply by mennatallah-abdallah-e** [reply]
what have you done ?

**Reply by lisinan88** [reply]
> what have you done ?

token and url should be corect ,by the way ,you need x or 2xlarge instance , do you use amazon linux ?or you local server?

**Reply by lisinan88** [reply]
Do you know how to interageted other plugin ?

**Reply by mennatallah-abdallah-e** [reply]
I use AWS EKS. I changed some configuration. I have another question now. is cost analysis and budget working wiz you ? as for me it shows a contact us page. Would you please advise.

**Reply by lisinan88** [reply]
> I use AWS EKS. I changed some configuration. I have another question now. is cost analysis and budget working wiz you ? as for me it shows a contact us page. Would you please advise.

not working wiz me , i didnot find solution to resolved this issue

**Reply by mennatallah-abdallah-e** [reply]
I changed the spaceone configuration as per this 
https://cloudforet.io/docs/setup_operation/configuration/change_k8s_namespace/#change-the-namespace-of-plugin-service

and I have made the plugin and the spaceone services run in same namespace. 
and i made sure the rbac step https://cloudforet.io/docs/setup_operation/installation/aws/#3-create-role-and-rolebinding is made for spaceone.

**Reply by Siddhesh-Shinde** [reply]
@mennatallah-abdallah-e How did you fix this issue? Can you please elaborate?

**Reply by Siddhesh-Shinde** [reply]
@lisinan88 How did you fix this issue? Can you please elaborate?

**Reply by Siddhesh-Shinde** [reply]
I checked and my port 8082 doesn't seem to be working after the setup. I followed https://cloudforet.io/v1-12/docs/setup_operation/quick_install/
![image](https://github.com/user-attachments/assets/059a2956-c798-40b8-aa82-a1abdea0b617)

**Comment by mennatallah-abdallah-e** [comment]
Does Anyone has the cost analysis and budget working  ? as for me it shows a contact us page. Would you please advise.

**Reply by choonho** [reply]
Enable Cost Analysis can not be done by web console. You need to enable by API (or CLI).

Check my blog 

* AWS Billing
https://dev.to/choonho/cloudforet-enable-aws-billing-plugin-3oma

* Azure Billing
https://dev.to/choonho/cloudforet-enable-azure-billing-plugin-41ln

Please visit Discord, if you have more question
https://discord.gg/7ExpTmA6TE

**Reply by mennatallah-abdallah-e** [reply]
What do you mean by "We assume that you have configured spacectl API token already." I am working on AWS plugin. 

Thanks in advance.

**Reply by choonho** [reply]
To use spacectl cli tool, you need API token like awscli tool.

See the example in 
https://github.com/orgs/cloudforet-io/discussions/139

**Reply by mennatallah-abdallah-e** [reply]
I am running now the spacectl exec register cost_analysis.DataSource -f aws_data_source.yaml it is taking time. Thanks !

**Reply by mennatallah-abdallah-e** [reply]
How to fix this 
root@spacectl-54b4d5d5cb-zm8vr:~# spacectl exec register cost_analysis.DataSource -f aws_data_source.yaml
ERROR: install plugin is failed by timeout, supervisor_id: supervisor-4353eb69256c, plugin_id: plugin-aws-cost-explorer-cost-datasource, version: 1.0.6

**Reply by choonho** [reply]
There are some use-cases.

1) check other plugin PODs are deployed well, kubectl get pod -n cloudforet-plugin ( ex plugin-aws-ec2-collector)
2) Slow internet, "enable plugin" means deployment of new kubernetes POD. If downloading plugin POD image is too slow, we will cancel to deploy it. Try again (there is a docker image cache)
3) See the log of supervisor POD with trying again. kubectl logs -f supervisor-XXXXXX(check your exact pod name) -n cloudforet.

**Reply by choonho** [reply]
> There are some use-cases.
> 
> 1. check other plugin PODs are deployed well, kubectl get pod -n cloudforet-plugin ( ex plugin-aws-ec2-collector)
> 2. Slow internet, "enable plugin" means deployment of new kubernetes POD. If downloading plugin POD image is too slow, we will cancel to deploy it. Try again (there is a docker image cache)
> 3. See the log of supervisor POD with trying again. kubectl logs -f supervisor-XXXXXX(check your exact pod name) -n cloudforet.

If you use minikube, the CPU and Memory resource may be very small, it is hard to deploy plugin POD. You needs more kubernetes worker node.

**Reply by mennatallah-abdallah-e** [reply]
Hi @choonho 
I have a question regarding this article https://dev.to/choonho/cloudforet-enable-azure-billing-plugin-41ln . Do I have to create datasource to show the cost of each azure account or what is the billing_account_id ?

**Reply by choonho** [reply]
> Hi @choonho I have a question regarding this article https://dev.to/choonho/cloudforet-enable-azure-billing-plugin-41ln . Do I have to create datasource to show the cost of each azure account or what is the billing_account_id ?

Could you ask in Cloudforet Discord channel, https://discord.gg/7ExpTmA6TE , I can reply quickly in discord.

**Reply by choonho** [reply]
@mennatallah-abdallah-e enable billing data source is one-time job if your want single company billing data like enterprise agreement.

https://learn.microsoft.com/en-us/azure/cost-management-billing/manage/view-all-accounts

**Reply by mennatallah-abdallah-e** [reply]
Thanks @choonho I am working on that. is there any existing plugin integration with Azure advisor?

**Reply by choonho** [reply]
You can easily check by https://github.com/orgs/cloudforet-io/repositories?q=azure
But some plugins may not be compatible with your installation version (1.12)

For example, [plugin-azure-identity-account-collector](https://github.com/cloudforet-io/plugin-azure-identity-account-collector) is for Cloudforet ver 2.x which is active development version.

**Reply by lisinan88** [reply]
> Enable Cost Analysis can not be done by web console. You need to enable by API (or CLI).
> 
> Check my blog
> 
> * AWS Billing
>   https://dev.to/choonho/cloudforet-enable-aws-billing-plugin-3oma
> * Azure Billing
>   https://dev.to/choonho/cloudforet-enable-azure-billing-plugin-41ln
> 
> Please visit Discord, if you have more question https://discord.gg/7ExpTmA6TE

Hello ,how can i get  the data from aws resouce ,for me  it is empty

**Reply by lisinan88** [reply]
![image](https://github.com/cloudforet-io/cloudforet/assets/129405455/fb49337a-f747-4a4b-9d34-d407a7ee9fa3)
Does some one know ,why it is empty

**Reply by lisinan88** [reply]
i got this resolved
